inc/alert.class.php

```diff
@@ -210,79 +210,126 @@ public static function findAllToNotify($params = []) {
+                  ]
+               ],
+               [
+                  'AND' => [
+                     "$ttable.itemtype" => 'Group',
+                     "$ttable.items_id" => $fndgroup,
+                  ]
+               ],
+               [
+                  'AND' => [
+                     "$ttable.itemtype" => 'User',
+                     "$ttable.items_id" => $_SESSION['glpiID'],
+                  ]
+               ],
+            ]
+         ];
       } else if ($p['show_only_login_alerts']) {
-         $login_sql = " AND `$table`.`is_displayed_onlogin` = '1'";
+         $login_sql = ["{$table}.is_displayed_onlogin" => 1];
       }
 
       if ($p['show_hidden_alerts']) {
          //dont show hidden alert if they should no longer be visible
-         $login_show_hidden_sql = " `$utable`.`id` IS NOT NULL";
+         $login_show_hidden_sql = ['NOT' => ["{$utable}.id" => null]];
       }
 
       if ($p['show_only_central_alerts']) {
          //dont show central alert if they should no longer be visible
-         $show_central_sql = " AND `$table`.`is_displayed_oncentral`='1'";
+         $show_central_sql = ["{$table}.is_displayed_oncentral" => 1];
       }
 
       //If the alert must be displayed on helpdesk form : filter by ticket's entity
       //and not the current entity
       if ($p['show_only_helpdesk_alerts']) {
-         $show_helpdesk_sql = " AND `$table`.`is_displayed_onhelpdesk`='1'";
+         $show_helpdesk_sql = ["{$table}.is_displayed_onhelpdesk" => 1];
       }
       if (!$p['show_only_login_alerts']) {
-         $entity_sql = getEntitiesRestrictRequest("AND", $table, "", $p['entities_id'], true);
+         $entity_sql = getEntitiesRestrictCriteria($table, '', $p['entities_id'], true);
       }
-
-      $query = "SELECT DISTINCT `$table`.`id`, `$table`.*
-                  FROM `$table`
-                  LEFT JOIN `$utable`
-                     ON `$utable`.`plugin_news_alerts_id` = `$table`.`id`
-                     AND `$utable`.`users_id` = $users_id
-                     AND `$utable`.`state` = $hidstate
-                  INNER JOIN `$ttable`
-                     ON `$ttable`.`plugin_news_alerts_id` = `$table`.`id`
-                  $targets_sql
-                  WHERE ($login_show_hidden_sql $login_sql $show_central_sql $show_helpdesk_sql)
-                     AND (`$table`.`date_start` < '$today'
-                           OR `$table`.`date_start` = '$today'
-                           OR `$table`.`date_start` IS NULL
-                     )
-                     AND (`$table`.`date_end` IS NULL
-                           OR `$table`.`date_end` > '$today'
-                           OR `$table`.`date_end` = '$today'
-                     )
-                  AND `is_deleted`='0' AND `is_active`='1'
-                  $entity_sql";
-      $iterator = $DB->request($query);
-      if ($iterator->numrows() < 1) {
+      $criteria = [
+         'SELECT' => ["$table.*"],
```

Comment:
There was a `DISTINCT` keyword in previous code.
```suggestion
         'SELECT' => ["$table.*"],
         'DISTINCT' => true,
```